src/handler.rs

```diff
@@ -0,0 +1,199 @@
+    ) -> Result<(), failure::Error> {
+        let schema = BtcAnchoringSchema::new(self.context.snapshot());
+        let latest_anchored_height = schema.latest_anchored_height();
+        let anchoring_height = self
+            .anchoring_state
+            .following_anchoring_height(latest_anchored_height);
+
+        if self.context.height() < anchoring_height {
+            return Ok(());
+        }
+
+        // Creates anchoring proposal
+        let (proposal, proposal_inputs) =
+            if let Some(proposal) = schema.proposed_anchoring_transaction(&self.anchoring_state) {
+                proposal?
+            } else {
+                return Ok(());
+            };
+
+        let config = self.anchoring_state.actual_configuration();
+        let redeem_script = config.redeem_script();
+        // Creates Signature transactions.
+        let pubkey = redeem_script.content().public_keys[validator_id.0 as usize];
+        let mut signer = p2wsh::InputSigner::new(redeem_script);
+
+        for (index, proposal_input) in proposal_inputs.iter().enumerate() {
+            let input_id = TxInputId::new(proposal.id(), index as u32);
+
+            if let Some(input_signatures) = schema.transaction_signatures().get(&input_id) {
+                if input_signatures.contains(validator_id) {
+                    trace!(
+                        " {:?} is already signed by validator {}",
+                        input_id,
+                        validator_id
+                    );
+                    continue;
+                }
+            }
+
+            let signature = signer.sign_input(
+                TxInRef::new(proposal.as_ref(), index),
+                proposal_inputs[index].as_ref(),
+                privkey.0.secret_key(),
+            )?;
+
+            signer
+                .verify_input(
+                    TxInRef::new(proposal.as_ref(), index),
+                    proposal_input.as_ref(),
+                    &pubkey,
+                    &signature,
+                ).unwrap();
+
+            let signature_tx = Signature::new(
+                self.context.public_key(),
+                validator_id,
+                proposal.clone(),
+                index as u32,
+                signature.as_ref(),
+                self.context.secret_key(),
+            );
+
+            self.context
+                .transaction_sender()
+                .send(Box::new(signature_tx))?;
+        }
+
+        Ok(())
+    }
+
+    fn handle_as_auditor(self) -> Result<(), failure::Error> {
+        // TODO Think about corresponding business logic.
+        Ok(())
+    }
+}
+
+/// The goal of this task is to push uncommitted anchoring transactions to the bitcoin blockchain.
+#[derive(Debug)]
+pub struct SyncWithBtcRelayTask<'a> {
+    context: &'a ServiceContext,
+    relay: &'a dyn BtcRelay,
+}
+
+impl<'a> SyncWithBtcRelayTask<'a> {
+    /// Creates sync task instance for the given context and the bitcoin rpc relay.
```

Comment:
Creates synchronization task instance for the given context and the Bitcoin RPC relay.